crates/biome_js_analyze/src/lint/nursery/no_playwright_useless_await.rs

```diff
@@ -0,0 +1,362 @@
+use biome_rowan::{AstNode, BatchMutationExt, TokenText};
+
+use crate::JsRuleAction;
+
+declare_lint_rule! {
+    /// Disallow unnecessary `await` for Playwright methods that don't return promises.
+    ///
+    /// Some Playwright methods are frequently, yet incorrectly, awaited when they return
+    /// synchronous values. This includes locator methods, which return locators (not promises),
+    /// and synchronous expect matchers.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// await page.locator('.my-element');
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// await page.getByRole('button');
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// await expect(1).toBe(1);
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// page.locator('.my-element');
+    /// await page.locator('.my-element').click();
+    /// ```
+    ///
+    /// ```js
+    /// page.getByRole('button');
+    /// await page.getByRole('button').click();
+    /// ```
+    ///
+    /// ```js
+    /// expect(1).toBe(1);
+    /// await expect(page.locator('.foo')).toBeVisible();
+    /// ```
+    ///
+    pub NoPlaywrightUselessAwait {
+        version: "next",
+        name: "noPlaywrightUselessAwait",
+        language: "js",
+        sources: &[RuleSource::EslintPlaywright("no-useless-await").same()],
+        recommended: false,
+        fix_kind: FixKind::Safe,
+        domains: &[RuleDomain::Playwright],
+    }
+}
+
+// Locator methods that return Locator (synchronous)
+const LOCATOR_METHODS: &[&str] = &[
+    "and",
+    "first",
+    "getByAltText",
+    "getByLabel",
+    "getByPlaceholder",
+    "getByRole",
+    "getByTestId",
+    "getByText",
+    "getByTitle",
+    "last",
+    "locator",
+    "nth",
+    "or",
+];
+
+// Page/Frame methods that are synchronous
+const SYNC_PAGE_METHODS: &[&str] = &[
+    "childFrames",
+    "frame",
+    "frameLocator",
+    "frames",
+    "isClosed",
+    "isDetached",
+    "mainFrame",
+    "name",
+    "on",
+    "page",
+    "parentFrame",
+    "setDefaultNavigationTimeout",
+    "setDefaultTimeout",
+    "url",
+    "video",
+    "viewportSize",
+    "workers",
+];
+
+// Synchronous expect matchers (not Playwright-specific web-first assertions)
+const SYNC_EXPECT_MATCHERS: &[&str] = &[
+    "toBe",
+    "toBeCloseTo",
+    "toBeDefined",
+    "toBeFalsy",
+    "toBeGreaterThan",
+    "toBeGreaterThanOrEqual",
+    "toBeInstanceOf",
+    "toBeLessThan",
+    "toBeLessThanOrEqual",
+    "toBeNaN",
+    "toBeNull",
+    "toBeTruthy",
+    "toBeUndefined",
+    "toContain",
+    "toContainEqual",
+    "toEqual",
+    "toHaveLength",
+    "toHaveProperty",
+    "toMatch",
+    "toMatchObject",
+    "toStrictEqual",
+    "toThrow",
+    "toThrowError",
+];
+
+impl Rule for NoPlaywrightUselessAwait {
+    type Query = Ast<JsAwaitExpression>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = ();
```

Comment:
It seems the PR got old. You should add a file named like the file rule inside the crate `biome_rules_options`, and use that struct here. Refer to the existing code. It must be applied to all rules